### PR TITLE
add query params for roles in login page

### PR DESCRIPTION
**Pull request by 98ZhaoJeffrey**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/7e51fde5f22d448b8226f2cfa4dd93d5?v=b2dc0dba06574651a8b6063bdd0d74e0&p=d47d2e72e0854c998155b0f491280e95&pm=s)

## Implementation description
* Included role types within the login page accepting only administrators, learners and facilitators to login
* Only allow admins and learners to login, facilitators must signup
* Redirects to welcome page if role doesn't exist or is blank

## Steps to test
1. Go to /login and check that you are redirected to /welcome
2. Try /login?role={role} where role here is NOT any of administrator, learner and facilitator (case sensitive) -> should be on /welcome as well
3. Now try /login?role=facilitator, you should only be able to sign up
4. Clicking on signup should redirect to /signup?role=facilitator
5. Now try /login?role={role} where role here is administrator or learner. 
6. The form to login should be here and the sign up button should work the same as for facilitator with the corresponding role
7. For now, change the user in mongodb to have role learner
8. Try logging in under /login?role=learner, this should succeed
9. Try logging in under /login?role=administrator, this should fail and alert you in browser

## What should reviewers focus on?
* Check if the roles displays the correct UI and that login checks the role correctly.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/auth/Login.tsx [hunk 1]
```diff
@@ -24,13 +25,39 @@ const Login = (): React.ReactElement => {
...
+      alert(`Bad login. Expected ${user.role}, got ${role}`);
+      return;
+    }
+    localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(user));
     setAuthenticatedUser(user);
   };
 
   const onSignupClick = () => {
-    history.push(SIGNUP_PAGE);
+    history.push(`${SIGNUP_PAGE}?role=${role}`);
```
Comment (carolynzhang18) [inline frontend/src/components/auth/Login.tsx:60]: u are awesome for thinking ahead on this thank u!!

**Review by carolynzhang18 — APPROVED** [review]
LGTM!!! Everything works as expected, thanks for making those fixes!!! 😎

**Comment by cherryyng** [comment]
Everything looks good! All the testing worked as expected!

**Review by cherryyng — APPROVED** [review]
Looks good!

**Pull request merged**